metagpt/provider/__init__.py

```diff
@@ -5,20 +5,36 @@
 @Author  : alexanderwu
 @File    : __init__.py
 """
+import importlib
+class LLMFactory:
+    def __init__(self, module_name, instance_name):
+        self.module_name = module_name
+        self.instance_name = instance_name
+        self._module = None
 
-from metagpt.provider.google_gemini_api import GeminiLLM
-from metagpt.provider.ollama_api import OllamaLLM
-from metagpt.provider.openai_api import OpenAILLM
-from metagpt.provider.zhipuai_api import ZhiPuAILLM
-from metagpt.provider.azure_openai_api import AzureOpenAILLM
-from metagpt.provider.metagpt_api import MetaGPTLLM
-from metagpt.provider.human_provider import HumanProvider
-from metagpt.provider.spark_api import SparkLLM
-from metagpt.provider.qianfan_api import QianFanLLM
-from metagpt.provider.dashscope_api import DashScopeLLM
-from metagpt.provider.anthropic_api import AnthropicLLM
-from metagpt.provider.bedrock_api import BedrockLLM
-from metagpt.provider.ark_api import ArkLLM
+    def __getattr__(self, name):
+        if self._module is None:
+            self._module = importlib.import_module(self.module_name)
+        return getattr(self._module, name)
+    def __instancecheck__(self, instance):
+        if self._module is None:
+            self._module = importlib.import_module(self.module_name)
+        return isinstance(instance, getattr(self._module, self.instance_name))
+
+    
+GeminiLLM = LLMFactory("metagpt.provider.google_gemini_api ", "GeminiLLM")
+OllamaLLM = LLMFactory("metagpt.provider.ollama_api ", "OllamaLLM")
+OpenAILLM = LLMFactory("metagpt.provider.openai_api ", "OpenAILLM")
```

Comment:
It can't support usage like below
```
from metagpt.config2 import config
from metagpt.provider import OpenAILLM

llm_config = config.llm
llm = OpenAILLM(llm_config)
```